apps/mull-api/src/app/user/user.service.ts

```diff
@@ -19,16 +21,23 @@ export class UserService {
     return this.userRepository.findOne(id);
   }
 
+  findByEmail(email: string): Promise<User[]> {
+    return this.userRepository.find({ where: { email } });
+  }
+
   async findAllFriends(id: number): Promise<User[]> {
     const { friends } = await this.userRepository.findOne(id, { relations: ['friends'] });
     return friends;
   }
 
   async create(userInput: CreateUserInput): Promise<User> {
-    const { dob, name, email, password } = userInput;
-    const newUser = this.userRepository.create({ name, dob, email, password });
-    await this.userRepository.save(newUser);
-    return newUser;
+    if (userInput.type === UserType.LOCAL) {
```

Comment:
Sorry, missed this in my initial review.
I was able to register using an empty string as a password, so we should add a check for that here as well.

Reply:
Or hmm the frontend implementation can (and should) also do the form validation, so I am unsure of my comment now.

Reply:
I could add backend validation. I think we should start having something for it.

Reply:
The more validation the better

Reply:
Decided to use the class-validator package for validation as it was recommended by the nestjs docs.

Reply:
I can no longer create a user with an empty string as password, therefore marking this as resolved.